README.md

```diff
@@ -1,42 +1,211 @@
+set name=$topds
+end
+add fs
+set dir=/smartdc
+set special=/smartds
+set type=lofs
+set options=ro
+end
+add device
+set match=kvm
+end
+EOF
+```
 
- * SmartOS
- * A running Debian instance (physical or virtual) with spare disk space
- * sdc-vmtools
- 
 ## Setup
 
-Install live-build:
+This relies on the sdc-vmtools repo as a submodule.  You can get the right
+version of that with:
 
 ```
-apt-get install live-build
+git submodules update --init
 ```
 
-Put the sdc-vmtools directory in the root of this repository.
+If you forget to do that, `create-image` will do it before it tries to use
+anything from that submodule.
+
+## Using
+
+To generate a Debian `<version>` image run:
 
-## Configure
+```
+# ./create-image -r <version>
+```
 
-Modify the scripts in `auto\` and `config` depending on how you want your ISO configured. See the [live-build manual](http://live-systems.org/manual/stable/html/live-manual.en.html) for more information.
 
+```
+$ ./create-image -h
+Usage:
+        ./create-image [options] [command ...]
+option:
+        -h          This message
+        -r          Distro release
+
+Commands:
+        fetch       Fetch the installation ISO
+        iso         Create a kickstart ISO
+        image       Generate the image
+```
+
+### fetch
+
+Download the distribution's netinst media (.iso) and verify its integrity.
+If the required ISO already exists, its integrity is verified.  If it is found
+to be corrupt it is fetched again.
+
+This image will be automatically mounted at `/cdrom` during installation.
+
+### iso
+
+Generate an installer ISO image.  This will contain the following:
+
+* `preseed.cfg` - From `debian-<release>/preseed.cfg`.
+* `sdc-vmtools` - The current content of the
+  [sdc-vmtools](https://github.com/joyent/sdc-vmtools) repo.
+* `early-command` - A preinstallation script, from
+  `debian-<release>/early-command`
+* `late-command` - A postinstallation script, from
+  `debian-<release>/late-command`
+
+This image is automatically mounted when the debian installer needs it, but is
+not automtically mounted in the chroot (`/target`) while `late-command` is
+running.  For that reason, each release's late command needs to mount it at
+`/target/cdrom` prior to running `late-command`.
+
+
+### image
+
+This runs qemu in a way that allows unattended installation using the media
+described above.  Once qemu exits, a Triton-compatible
+image is generated and stored in the current `bits/<stamp>` directory as
+`debian-<release>-<timestamp>.{json,tar.gz}`.
+
+The actual image creation is handled by `sdc-vmutils/bin/create-hybrid-iamge`.
 
-## Using
 
-Once configured, run the following command:
+### upload
 
+This invokes
+[bits-upload.sh](https://github.com/joyent/eng/blob/master/tools/bits-upload.sh)
+to upload the image to manta and updates.joyent.com.
+
+
+## Default Settings For Images
+
+Each image has the following characteristics.  See
+`debian-<release>/ks.cfg` for details on which packages are included.
```

Comment:
Is `ks.cfg` the correct file for Debian (the name suggests something kickstart related)?

Reply:
s/ks/preseed/